### PR TITLE
java: make sure oauth gated properly

### PR DESCRIPTION
**Pull request by dcb6**
This change checks generator config to ensure that oauth is only generated if the user has permission. If not, we replace this auth scheme with a default bearer token auth scheme.

As part of this change, we now store a `resolvedAuthSchemes` in the context and read from this rather than IR in our code paths.

### CODE REVIEW

File: generators/java/sdk/src/main/java/com/fern/java/client/Cli.java [hunk 1]
```diff
@@ -198,19 +198,21 @@ public GeneratedRootClient generateClient(ClientGeneratorContext context, Interm
         generatedTypes.getTypes().values().forEach(this::addGeneratedFile);
         generatedTypes.getInterfaces().values().forEach(this::addGeneratedFile);
 
-        Optional<OAuthScheme> maybeOAuthScheme = context.getIr().getAuth().getSchemes().stream()
-                .map(AuthScheme::getOauth)
-                .flatMap(Optional::stream)
-                .findFirst();
+        Optional<GeneratedJavaFile> generatedOAuthTokenSupplier = Optional.empty();
+        if (context.getGeneratorConfig().getGenerateOauthClients()) {
+            Optional<OAuthScheme> maybeOAuthScheme = context.getIr().getAuth().getSchemes().stream()
```
Comment (dcb6) [inline generators/java/sdk/src/main/java/com/fern/java/client/Cli.java:203]: should i throw if someone has oauth configured but `context.getGeneratorConfig().getGenerateOauthClients()` is false?

as written, just doesn't generate the oauth stuff

Comment (dsinghvi) [reply generators/java/sdk/src/main/java/com/fern/java/client/Cli.java:203]: good question i think not generating is okay or falling back to bearer, @amckinney what do you do in the go / ts generators?

Comment (dcb6) [reply generators/java/sdk/src/main/java/com/fern/java/client/Cli.java:203]: looks to me like we don't throw in TS:
https://github.com/fern-api/fern/blob/main/generators/typescript/sdk/generator/src/SdkGenerator.ts#L382

couldn't find code falling back to bearer but not 100%, would need @amckinney to confirm

Comment (amckinney) [reply generators/java/sdk/src/main/java/com/fern/java/client/Cli.java:203]: Discussed offline - 

We'll go with falling back to bearer in all generators. We'll follow-up with test cases / additional handling in all of the generators that support OAuth.

File: generators/java/generator-utils/src/main/java/com/fern/java/FeatureResolver.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    public List<AuthScheme> getResolvedAuthSchemes() {
+        List<AuthScheme> schemes = ir.getAuth().getSchemes();
+        Optional<OAuthScheme> maybeOAuthScheme = schemes.stream()
+                .map(AuthScheme::getOauth)
+                .flatMap(Optional::stream)
+                .findFirst();
+        if (maybeOAuthScheme.isEmpty() || generatorConfig.getGenerateOauthClients()) return schemes;
+        generatorExecClient.sendUpdate(GeneratorUpdate.log(LogUpdate.builder()
+                .level(LogLevel.ERROR)
+                .message("OAuth is not supported in your current Java SDK plan. Please reach out to the Fern team!")
```
Comment (dcb6) [inline generators/java/generator-utils/src/main/java/com/fern/java/FeatureResolver.java:68]: Could use pointers on copy here

Comment (amckinney) [reply generators/java/generator-utils/src/main/java/com/fern/java/FeatureResolver.java:63]: We might also want to include a message that describes we're using bearer instead.
```suggestion
                .message("OAuth is not supported in your current Java SDK plan; falling back to bearer auth. Please reach out to the Fern team!")
```

**Review by amckinney — APPROVED** [review]
Looks great!

**Pull request merged**